Source: https://github.com/Dushistov/sdcv/pull/104.patch
--
From aec5081a8c852ef7f7330985472c564eed6c998c Mon Sep 17 00:00:00 2001
From: Andrew Savchenko <bircoph@altlinux.org>
Date: Fri, 20 Sep 2024 16:32:47 +0300
Subject: [PATCH] stardict_lib.cpp: use explicit const_cast

Gcc-14 considers incompatible pointer types as an error, so use
expcilict const_cast to convert from const char* to gchar*.

Signed-off-by: Andrew Savchenko <bircoph@altlinux.org>
---
 src/stardict_lib.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/stardict_lib.cpp b/src/stardict_lib.cpp
index 83fbc59..799a025 100644
--- a/src/stardict_lib.cpp
+++ b/src/stardict_lib.cpp
@@ -1047,7 +1047,7 @@ bool Libs::LookupSimilarWord(const gchar *sWord, std::set<glong> &iWordIndices,
         }
         // Upper the first character and lower others.
         if (!bFound) {
-            gchar *nextchar = g_utf8_next_char(sWord);
+            gchar *nextchar = const_cast<gchar*>(g_utf8_next_char(sWord));
             gchar *firstchar = g_utf8_strup(sWord, nextchar - sWord);
             nextchar = g_utf8_strdown(nextchar, -1);
             casestr = g_strdup_printf("%s%s", firstchar, nextchar);
